main.tf

```
@@ -39,8 +83,8 @@ data "aws_iam_policy_document" "service_account_assume_role" {
     }
 
     condition {
-      test     = "StringEquals"
-      values   = [format("system:serviceaccount:%s:%s", var.service_account_namespace, var.service_account_name)]
+      test     = local.service_account_namespace_provided == false || local.service_account_name_provided == false ? "StringLike" : "StringEquals"
```

We can use `StringLike` in all cases. It would only be a problem if one of the provided names had a wildcard, which I am not worried about since it would create an illegal role name.
```suggestion
      test     = "StringLike"
```
IAM requires you use `StringLike` for [exact matches of strings that have IAM policy variables in them](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html#Conditions_String) so I am confident it is OK for us to use it here.

---

we used `StringLike` before, then we switched to `StringEqual` (not sure why).
But yes, `StringLike` will works in all cases even if we don't need to match a wildcard.